python_modules/dagster/dagster/_core/definitions/asset_layer.py

```diff
@@ -675,6 +675,23 @@ def asset_info_for_output(
     ) -> Optional[AssetOutputInfo]:
         return self.asset_info_by_node_output_handle.get(NodeOutputHandle(node_handle, output_name))
 
+    def asset_check_handle_for_output(
+        self, node_handle: NodeHandle, output_name: str
+    ) -> Optional[AssetCheckHandle]:
+        check_names_by_asset_key = self.check_names_by_asset_key_by_node_handle.get(node_handle, {})
```

Comment:
maybe we should store a NodeOutputHandle -> CheckSpec mapping in AssetLayer instead of going digging?

Reply:
can optimize this in the future though, doesn't change what we're storing